src/evo/deterministicmns.cpp

```diff
@@ -27,6 +26,10 @@
 #include <optional>
 #include <memory>
 
+// Forward declaration to break dependency over node/transaction.h
```

Comment:
Can't say I'm particularly keen on repeated redefinition (any updates to the function name and arguments will require four additional updates on top of the header change), is this PR a dependency for another PR where these circular dependencies must be broken?